wordpress-mobile/WordPress-Android
# Issue/ri post card ui misc fixes

This PR is a subpart of #11963 and includes

- Remove Bookmarked Post Card UI (Saved tab) improvements
- Remove Follow button from Likes tab post cards (option available in more options)
- Fixes (because of constraint layout changes in the post card UI)
    - Likes tab: site name and url in the header are truncated, content should spread to available space
    - Thumbnails strip overlapping header when visible (CardType: [Gallery](https://github.com/wordpress-mobile/WordPress-Android/blob/50f98a16880a535af82c35f7a81c3c65e592fa9a/WordPress/src/main/java/org/wordpress/android/models/ReaderCardType.java#L43))
-----------

**To Test**

1. Bookmark few posts in the reader
2. Go to Saved tab
3. Un bookmark  few posts
4. Notice that removed bookmark post cards are displayed properly with a gutter space b/w them

![device-2020-05-22-114231](https://user-images.githubusercontent.com/1405144/82636945-6163b680-9c21-11ea-8b3d-bef6506660e2.png)

-----------

1.  Create a private post
2. Add at least 4 images (250px < width < 640px)
3. Like it
4. Go to Likes tab in the reader
5. Notice that 
    a. Header title spreads to available space
    b. Title is displayed below the gallery image
    c. There's no "follow" button next to header title. Follow option is available in more options.

![device-2020-05-22-113920](https://user-images.githubusercontent.com/1405144/82636744-ef8b6d00-9c20-11ea-8f0c-93fbad4d6e93.png)

PR submission checklist:
